components/sup/src/manager/service/mod.rs

```diff
@@ -471,12 +497,12 @@ impl Service {
                                       leader_id.to_string());
                             self.last_election_status = census_group.election_status;
                         }
-                        self.execute_hooks(launcher, executor, hook_update_table, config_change)
+                        self.execute_hooks(launcher, executor, &template_update)
                     }
                 }
             }
         }
-        if svc_updated {
+        if template_update.changed() {
```

Comment:
🤔 This may be too coarse, now... as coded here, this would retrigger health checks if, say, just the `init` hook were to change. Since that no longer will trigger a restart, there's no way it would affect the running service. You might be able to make an argument that health checks should be restarted if just the health check hook were to change, though.

Digging a bit further, it seems that `svc_updated` was _actually_ only indicating if any of the source data for templating had changed, and _not_ if the service was actually updated (reconfigured, restarted, actually changed in any way), as advertised. This ultimately bubbles up to cause an updated service rumor to be sent out, which is a bit surprising; why should we send out an updated service rumor if nothing has affected the running of the service at all?

(So, `template_update.changed()` is, strictly speaking, changing this behavior, but now I'm not convinced the previous behavior was correct, either 😱 )

Reply:
Interesting. I had a similar thoughts about the health check, but was confused by the comments in in that `if` block. They seem **almost** contradictory. On one hand, we want to run the health check right away after **any** change to immediately detect if there was a problem. On the other hand, we want to wait enough time to let the service get up to speed. One option would be to completely remove the restart? That way we are not trying to guess how long is long enough.

As for the return value, it sounds like we maybe want to change the behaviour even further, and only return `true` if the service was restarted or reconfigured (opposed to any change to hooks or config). Is there a good way to test if this breaks things?

Reply:
I think health checks may just naturally restart once a service restarts anyway (if they don't, then that seems like a more natural place to for this behavior, rather than dealing with extra special cases... though we would still want to take reconfigures into account).

As for changing the return value... yeah, I agree that we should change it. Testing will likely involve resurrecting our Kubernetes testing cluster, which is slated to be added into our testing pipelines at some point in the near future. (pinging @scotthain)

Reply:
@christophermaier We should be able to set up an e2e test with the kube cluster pretty easily once we get that pipeline hooked up. I've never used/seen the old kube cluster tho so I have very little context on its actual operation.